staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -89,12 +91,22 @@ func (c *Config) TransportConfig() (*transport.Config, error) {
 		Proxy: c.Proxy,
 	}
 
-	if c.ExecProvider != nil && c.AuthProvider != nil {
+	if c.Exec.ExecProvider != nil && c.AuthProvider != nil {
 		return nil, errors.New("execProvider and authProvider cannot be used in combination")
 	}
 
-	if c.ExecProvider != nil {
-		provider, err := exec.GetAuthenticator(c.ExecProvider)
+	if c.Exec.ExecProvider != nil {
+		caData, err := dataFromSliceOrFile(c.CAData, c.CAFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load CA bundle for execProvider: %v", err)
+		}
+		cluster := clientauthentication.Cluster{
```

Comment:
isn't this missing other bits like proxyURL and insecure? is there a unit test that should have caught this?

Reply:
https://github.com/kubernetes/kubernetes/pull/95489/commits/bd4af0df52b7da2a838795ccf9a8760a0316da48

whoops. yeah, you're right - we _should_ be passing through those other fields.

i am scratching my head for where we could put this unit test. i don't see a unit test in the `rest` package for this code. i also don't see any test higher up that verifies that the initialization of an exec plugin is done correctly.

i could add a unit test in `rest`, but it seems non-trivial to get the cluster info that we pass to `exec.GetAuthenticator`. i was tempted to add a test to `k8s.io/client-go/plugin/pkg/auth/exec`, but those unit tests are pretty self-contained to that package's functionality (which totally makes sense).

best i could come up with is to add a helper function that maps from `rest.Config` to `clientauthentication.Cluster`.

the missing test coverage that you called out still exists, because if i completely remove the call to the new helper function, then all the `rest` unit tests still pass. do you have any better ideas?